chef/cookbooks/cpe_chrome/resources/cpe_chrome.rb

```diff
@@ -27,7 +27,8 @@ def manage_chrome
   end
 
   def manage_chrome_osx
-    prefs = node['cpe_chrome'].reject { |_k, v| v.nil? }
+    prefs = node['cpe_chrome']['profile'].reject \
+      { |_k, v| v.nil? || v.class == Chef::Node::ImmutableArray && v.empty? }
```

Comment:
You are correct. I placed your default attributes file into chef shell. My error was the first `reject` which should be a select. I assume you want all arrays and drop empties which yields this fiddling in Chef shell:
```
chef:attributes (12.18.31)> node['cpe_chrome']['profile'].select{ |_k, v| v.is_a?(Chef::Node::ImmutableArray) }.reject{ |_k,v| v.empty? }
 => {}
chef:attributes (12.18.31)> default['cpe_chrome']['profile']['ExtensionInstallSources'] = ['a', 'b']
 => ["a", "b"]
chef:attributes (12.18.31)> node['cpe_chrome']['profile'].select{ |_k, v| v.is_a?(Chef::Node::ImmutableArray) }.reject{ |_k,v| v.empty? }
 => {"ExtensionInstallSources"=>["a", "b"]}
```

```
prefs = node['cpe_chrome']['profile'].
   select { |_k, v| v.is_a?(Chef::Node::ImmutableArray) }.
   reject { |_k, v| v.empty? }
```